python/py-hdbscan/Portfile

```diff
@@ -0,0 +1,76 @@
+
+long_description    HDBSCAN - Hierarchical Density-Based Spatial\
+                    Clustering of Applications with Noise. Performs\
+                    DBSCAN over varying epsilon values and integrates\
+                    the result to find a clustering that gives the\
+                    best stability over epsilon. This allows HDBSCAN\
+                    to find clusters of varying densities (unlike\
+                    DBSCAN), and be more robust to parameter\
+                    selection. In practice this means that HDBSCAN\
+                    returns a good clustering straight away with\
+                    little or no parameter tuning -- and the primary\
+                    parameter, minimum cluster size, is intuitive and\
+                    easy to select. HDBSCAN is ideal for exploratory\
+                    data analysis\; it's a fast and robust algorithm\
+                    that you can trust to return meaningful clusters\
+                    (if there are any).
+
+homepage            https://hdbscan.readthedocs.io/en/latest/
+
+checksums           rmd160  0059cef4f9a2a870a9d7c103ee4220bb8ad796a2 \
+                    sha256  29a07d47a4ff2d9e4938bb8d6f6b3c02e77b13669b0dc4cd3b2d43531c3aa957 \
+                    size    10431303
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools \
+                    port:py${python.version}-cython
+    depends_lib-append \
+                    port:libomp \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-scikit-learn \
+                    port:py${python.version}-scipy \
+                    port:py${python.version}-six
+    depends_run-append \
+                    port:py${python.version}-joblib
+    depends_test-append \
+                    port:py${python.version}-nose
+
+    compiler.openmp_version 2.5
+
+    python.add_archflags no
+
+    build.env-append    "CPPFLAGS=${configure.cppflags} -I${prefix}/include/libomp -Xpreprocessor -fopenmp" \
+                        "CFLAGS=${configure.cc_archflags} -I${prefix}/include/libomp" \
+                        "CXXFLAGS=${configure.cxx_archflags} -I${prefix}/include/libomp" \
+                        "LDFLAGS=${configure.ld_archflags} -L${prefix}/lib/libomp -lomp" \
+                        DYLD_LIBRARY_PATH=${prefix}/lib/libomp
+    destroot.env-append "CPPFLAGS=${configure.cppflags} -I${prefix}/include/libomp -Xpreprocessor -fopenmp" \
+                        "CFLAGS=${configure.cc_archflags} -I${prefix}/include/libomp" \
+                        "CXXFLAGS=${configure.cxx_archflags} -I${prefix}/include/libomp" \
+                        "LDFLAGS=${configure.ld_archflags} -L${prefix}/lib/libomp -lomp" \
+                        DYLD_LIBRARY_PATH=${prefix}/lib/libomp
+
+    test.run        yes
+    test.env        PYTHONPATH=${worksrcpath}/build/lib
```

Comment:
since the port is compiling extensions the directory ```${worksrcpath}/build/lib``` does not exist (it's something like ```lib.macosx-10.14-x86_64-3.8``` depending on your system/Python version). In these cases you should use (as we've seen in one of your other PRs:
```
    pre-test {
        test.env    PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]
    }
```
However, here since upstream supports -and you're using- the default set by the PG (i.e., ```{python.bin} setup.py --no-user-cfg test```), you do not need to specify ```PYTHONPATH```.

So since it's wrong and actually not needed anyway, please remove this line (perhaps you can align the "yes" with 4 more spaces as in the surrounding code).

Reply:
Thanks done.